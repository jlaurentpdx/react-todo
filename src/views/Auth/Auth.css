.auth-form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

label,
input,
button {
  display: block;
  margin: 5px;
  width: 10vw;
  min-width: 150px;
  max-width: 230px;
}

button {
  margin: 1%;
  color: white;
  background-color: darkslateblue;
}
