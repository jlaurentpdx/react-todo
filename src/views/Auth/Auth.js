import { useState } from 'react';
import * as users from '../../services/users';
import Header from '../../components/Header/Header';
import AuthForm from '../../components/AuthForm/AuthForm';
import './Auth.css';

export default function Auth({ setCurrentUser }) {
  const [type, setType] = useState('login');
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');
  const [message, setMessage] = useState('');

  const handleSubmit = async (e) => {
    let response;
    try {
      e.preventDefault();
      response =
        type === 'login'
          ? await users.signInUser(email, password)
          : await users.signUpUser(email, password);
      setMessage('Success! Redirecting...');
      setTimeout(() => {
        setCurrentUser(response);
      }, 1000);
    } catch {
      setMessage('Invalid email or password');
    }
  };

  return (
    <>
      <div>
        <Header {...{ type, setType, setMessage }} />
        <AuthForm {...{ email, setEmail, password, setPassword, message, handleSubmit }} />
      </div>
    </>
  );
}
