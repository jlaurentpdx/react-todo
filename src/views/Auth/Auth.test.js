import { render, screen } from '@testing-library/react';
import Auth from './Auth';

test('Auth should display Header and AuthForm when user is not logged in', () => {
  const { container } = render(<Auth setCurrentUser={null} />);

  screen.getByText('Login');
  screen.getByText('Register');
  screen.getAllByDisplayValue('');
  screen.getByPlaceholderText('email@example.com');
  screen.getByPlaceholderText('password');
  screen.getByText('Submit');

  expect(container).toMatchSnapshot();
});
