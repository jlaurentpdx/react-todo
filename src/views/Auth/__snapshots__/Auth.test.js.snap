// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Auth should display Header and AuthForm when user is not logged in 1`] = `
<div>
  <div>
    <header>
      <h1
        class="active"
      >
        Login
      </h1>
      <h1
        class=""
      >
        Register
      </h1>
    </header>
    <div>
      <p
        style="color: rgb(255, 0, 0);"
      />
      <form
        class="auth-form"
      >
        <label
          for="email"
        >
          E-mail:
        </label>
        <input
          name="email"
          placeholder="email@example.com"
          type="email"
          value=""
        />
        <label
          for="password"
        >
          Password:
        </label>
        <input
          name="password"
          placeholder="password"
          type="password"
          value=""
        />
        <button>
          Submit
        </button>
      </form>
    </div>
  </div>
</div>
`;
